pkg/boot/esxi/esxi.go

```diff
@@ -93,16 +108,30 @@ func getImages(device string, opts5, opts6 *options) ([]*boot.MultibootImage, er
 // LoadCDROM loads an ESXi multiboot kernel from a CDROM at device.
 //
 // device will be mounted at mountPoint.
-func LoadCDROM(device string) (*boot.MultibootImage, error) {
+func LoadCDROM(device string) (*boot.MultibootImage, *mount.MountPoint, error) {
 	mountPoint, err := ioutil.TempDir("", "esxi-mount-")
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	if _, err := mount.Mount(device, mountPoint, "iso9660", "", unix.MS_RDONLY|unix.MS_NOATIME); err != nil {
-		return nil, err
+	mp, err := mount.Mount(device, mountPoint, "iso9660", "", unix.MS_RDONLY|unix.MS_NOATIME)
+	if err != nil {
+		os.RemoveAll(mountPoint)
```

Comment:
Instead of repeating `os.RemoveAll` at each return, could you do:

```
defer func() {
   if err != nil {
       os.RemoveAll(mountPoint)
   }
}()
```

The `err` variable is a reference. Same for `mp.Unmount`.

Reply:
I kinda like it explicit like this. And doing this correctly requires names returns, which I kind of dislike.